### ISSUE TITLE
`gopass list --flat` doesn't show secrets with the same names as folders

### ISSUE THREAD

**Issue by duxovni** [gopasspw/gopass#2338]
### Summary
If a secret has the same name as a folder, it won't be listed by `gopass list --flat`.

### Steps To Reproduce

```
gopass add foo/bar
gopass add foo/bar/baz
gopass list --flat
```

### Expected behavior

Should return:
```
foo/bar
foo/bar/baz
```

### Environment

- OS: NixOS
- OS version: 22.11
- gopass Version: 1.14.5
- Installation method: Nixpkgs

**Comment by dominikschulz** [comment]
Small but tricky little issue. I think I found it, but I'll need to make sure all the tests pass and nothing breaks.

**Comment by AnomalRoil** [comment]
I won't say it's by design, but note we might want to change the text around shadowing here if we change the behaviour:
https://github.com/gopasspw/gopass/blob/master/docs/commands/list.md#shadowing

**Comment by dominikschulz** [comment]
You are right. Thank you. I'll look at that once I get to fix the failing tests.

**Comment by duxovni** [comment]
This doesn't seem to have been fixed in 1.14.9? I'm seeing that `gopass list` now adds `(shadowed)` after shadowed entries, but shadowed entries are still missing from `gopass list --flat`.

**Comment by dominikschulz** [comment]
I'm confused. This should have been fixed before. #2341 did fix some cases but your PR looks legit.
I'll need a bit to think a about these edge cases.
